src/main/java/org/broadinstitute/dsm/cf/Covid19OrderRegistrarFunction.java

```diff
@@ -0,0 +1,152 @@
+
+import java.sql.Connection;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.annotations.SerializedName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.typesafe.config.Config;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.broadinstitute.dsm.careevolve.Authentication;
+import org.broadinstitute.dsm.careevolve.Covid19OrderRegistrar;
+import org.broadinstitute.dsm.careevolve.OrderResponse;
+import org.broadinstitute.dsm.careevolve.Patient;
+import org.broadinstitute.dsm.careevolve.Provider;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.DdpKit;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.elasticsearch.client.RestHighLevelClient;
+
+public class Covid19OrderRegistrarFunction  implements BackgroundFunction<Covid19OrderRegistrarFunction.OrderPayload> {
+
+    private static final Logger logger = Logger.getLogger(TestBostonKitTrackerDispatcher.class.getName());
+
+    @Override
+    public void accept(OrderPayload orderPayload, Context context) throws Exception {
+        logger.info("Processing request " + orderPayload);
+        // check first to see if it's already ordered.
+
+        Config cfg = CFUtil.loadConfig();
+        String careEvolveAccount = cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_ACCOUNT);
+        String careEvolveSubscriberKey = cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_SUBSCRIBER_KEY);
+        String careEvolveServiceKey = cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_SERVICE_KEY);
+        String careEvolveOrderEndpoint = cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_ORDER_ENDPOINT);
+        Authentication auth = new Authentication(careEvolveSubscriberKey, careEvolveServiceKey);
+        Provider provider = new Provider(cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_PROVIDER_FIRSTNAME),
+                cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_PROVIDER_LAST_NAME),
+                cfg.getString(ApplicationConfigConstants.CARE_EVOLVE_PROVIDER_NPI));
+
+        PoolingDataSource<PoolableConnection> dataSource = CFUtil.createDataSource(5, cfg.getString(DSM_DB_URL));
+
+        DDPInstance ddpInstance = null;
+
+        try (Connection conn = dataSource.getConnection()) {
+            ddpInstance = DDPInstance.getDDPInstanceWithRole(conn,"testboston", DBConstants.HAS_KIT_REQUEST_ENDPOINTS);
```

Comment:
is there a reason for checking that role? this is the role for checking dss for kits every hour, this is also the role we sometimes remove if we want to stop dsm from asking for new kits! if we need to check if the study does kit shipping or not `kit_request_activated` would be the safer choice!

Reply:
@pegahtah if the role is not used, please change it to just getDDPInstance